### PR TITLE
Feature/nfckt 712

### PR DESCRIPTION
**Pull request by alex-kontcur**
1. There should be a function in pharmacy service that creates subscriptions for all KonnektorConfigs the connector

2. There should be a function in the pharmacy service that unsubscribes all KonnektorConfigs

3. The ids of the subscriptions for every konnektor config should be saved in the config folder of this Konnektor

4. The unsubscribe and subscribe function should be callable by the REST services PharmacyResource

### CODE REVIEW

File: src/test/java/health/ere/ps/service/idp/client/IdpClientTest.java [hunk 1]
```diff
@@ -29,8 +14,23 @@
...
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.security.Security;
+import java.security.cert.X509Certificate;
+import java.util.logging.LogManager;
+import java.util.logging.Logger;
 
 @QuarkusTest
+@Disabled("TODO")
```
Comment (ManuelB) [inline src/test/java/health/ere/ps/service/idp/client/IdpClientTest.java:33]: Why do you disable this test case?

File: src/test/java/health/ere/ps/service/connector/certificate/CardCertificateReaderServiceTest.java [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
-import health.ere.ps.service.connector.cards.ConnectorCardsService;
-import io.quarkus.test.junit.QuarkusTest;
-import io.quarkus.test.junit.TestProfile;
+import java.io.IOException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.logging.LogManager;
 
 @QuarkusTest
+@Disabled("TODO")
```
Comment (ManuelB) [inline src/test/java/health/ere/ps/service/connector/certificate/CardCertificateReaderServiceTest.java:23]: Why do you disable this whole test?

File: src/test/java/health/ere/ps/service/connector/SingleConnectorServicesProviderTest.java [hunk 3]
```diff
@@ -73,12 +73,14 @@ public void testGetKeyFromBase64OctetDataUri() throws URISyntaxException, KeySto
     }
     
     @Test
+    @Disabled("TODO")
```
Comment (ManuelB) [inline src/test/java/health/ere/ps/service/connector/SingleConnectorServicesProviderTest.java:76]: This should still work.

File: src/test/java/health/ere/ps/service/connector/SingleConnectorServicesProviderTest.java [hunk 4]
```diff
@@ -1,29 +1,28 @@
...
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
 
 public class SingleConnectorServicesProviderTest {
     private String keystoreFileName = "./src/test/resources/certs/keystore.p12";
-    private String keystoreFilePass = "123456";
+    private String keystoreFilePass = "123456.properties";
```
Comment (ManuelB) [inline src/test/java/health/ere/ps/service/connector/SingleConnectorServicesProviderTest.java:25]: Why do you change the keystoreFilePass?

File: src/test/java/health/ere/ps/resource/gematik/ERezeptWorkflowResourceTest.java [hunk 5]
```diff
@@ -75,11 +73,11 @@ public void testIdpToken() {
         eRezeptWorkflowResource.httpServletRequest = mock(HttpServletRequest.class);
         when(eRezeptWorkflowResource.httpServletRequest.getHeaderNames()).thenReturn(Collections.enumeration(Collections.emptyList()));
         eRezeptWorkflowResource.eRezeptWorkflowService = mock(ERezeptWorkflowService.class);
-        when(eRezeptWorkflowResource.eRezeptWorkflowService.getBearerToken(any())).thenReturn("123456");
+        when(eRezeptWorkflowResource.eRezeptWorkflowService.getBearerToken(any())).thenReturn("123456.properties");
```
Comment (ManuelB) [inline src/test/java/health/ere/ps/resource/gematik/ERezeptWorkflowResourceTest.java:76]: Why do you add .properties to 123456?

File: src/test/java/health/ere/ps/config/UserConfigTest.java [hunk 6]
```diff
@@ -30,7 +29,7 @@ public void setup() {
     public void testGetConfigurations() {
         UserConfigurations configurations = userConfig.getConfigurations();
         assertEquals("https://example.com", configurations.getConnectorBaseURL());
-        assertEquals("123456", configurations.getMandantId());
+        assertEquals("123456.properties", configurations.getMandantId());
```
Comment (ManuelB) [inline src/test/java/health/ere/ps/config/UserConfigTest.java:32]: There should not be any .properties

**Review by ManuelB — CHANGES_REQUESTED** [review]

File: src/test/java/health/ere/ps/service/connector/SingleConnectorServicesProviderTest.java (hunk 4 above)
Comment (alex-kontcur) [reply src/test/java/health/ere/ps/service/connector/SingleConnectorServicesProviderTest.java:25]: oops, this is autoreplacement.. reverting

File: src/test/java/health/ere/ps/service/idp/client/IdpClientTest.java (hunk 1 above)
Comment (alex-kontcur) [reply src/test/java/health/ere/ps/service/idp/client/IdpClientTest.java]: Reverted

File: src/test/java/health/ere/ps/service/connector/certificate/CardCertificateReaderServiceTest.java (hunk 2 above)
Comment (alex-kontcur) [reply src/test/java/health/ere/ps/service/connector/certificate/CardCertificateReaderServiceTest.java]: Reverted

File: src/test/java/health/ere/ps/resource/gematik/ERezeptWorkflowResourceTest.java (hunk 5 above)
Comment (alex-kontcur) [reply src/test/java/health/ere/ps/resource/gematik/ERezeptWorkflowResourceTest.java]: Reverted

File: src/test/java/health/ere/ps/config/UserConfigTest.java (hunk 6 above)
Comment (alex-kontcur) [reply src/test/java/health/ere/ps/config/UserConfigTest.java]: Reverted

**Review by ManuelB — APPROVED** [review]

**Pull request merged**